lib/matplotlib/axes/_axes.py

```diff
@@ -157,6 +163,14 @@ def set_title(self, label, fontdict=None, loc=None, pad=None, **kwargs):
         if loc is None:
             loc = rcParams['axes.titlelocation']
 
+        if y is None:
+            y = rcParams['axes.titley']
+        if y is None:
+            y = 1.0
```

Comment:
Do you need `self._autotitlepos = True` here?

Reply:
The problem here is that there are really three titles with one interface.  So setting `self._autotitlepos=True` here would allow toggling the automatic behaviour, but that behaviour would also affect other titles.  i.e.

```python
ax.set_title('Center', y=1.0)
ax.set_title('Left', loc='left')
```
would turn auto positioning on again.  

I think the current proposed behaviour is if you make it manual, it stays manual, leads to the least unexpected results.  We could make it all more complicated, but I really don't envision folks toggling this.

Reply:
Ok, that might still confuse someone, but you're right, the alternative would be more problematic because people might want to position the different titles differently.